waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -234,87 +216,88 @@ proc proofGen*(rlnInstance: RLN[Bn256], data: openArray[byte],
   if not proofIsSuccessful:
     return err("could not generate the proof")
 
-  var proofValue = cast[ptr array[416, byte]] (proof.`ptr`)
-  let proofBytes: array[416, byte] = proofValue[]
+  var proofValue = cast[ptr array[320, byte]] (proof.`ptr`)
+  let proofBytes: array[320, byte] = proofValue[]
   debug "proof content", proofHex = proofValue[].toHex
 
-  ## parse the proof as |zkSNARKs<256>|root<32>|epoch<32>|share_x<32>|share_y<32>|nullifier<32>|
+  ## parse the proof as [ proof<128> | root<32> | epoch<32> | share_x<32> | share_y<32> | nullifier<32> | rln_identifier<32> ]
+
   let
-    proofOffset = 256
+    proofOffset = 128
     rootOffset = proofOffset + 32
     epochOffset = rootOffset + 32
     shareXOffset = epochOffset + 32
     shareYOffset = shareXOffset + 32
     nullifierOffset = shareYOffset + 32
+    rlnIdentifierOffset = nullifierOffset + 32
 
   var
     zkproof: ZKSNARK
     proofRoot, shareX, shareY: MerkleNode
     epoch: Epoch
     nullifier: Nullifier
+    rlnIdentifier: RlnIdentifier
 
   discard zkproof.copyFrom(proofBytes[0..proofOffset-1])
   discard proofRoot.copyFrom(proofBytes[proofOffset..rootOffset-1])
   discard epoch.copyFrom(proofBytes[rootOffset..epochOffset-1])
   discard shareX.copyFrom(proofBytes[epochOffset..shareXOffset-1])
   discard shareY.copyFrom(proofBytes[shareXOffset..shareYOffset-1])
   discard nullifier.copyFrom(proofBytes[shareYOffset..nullifierOffset-1])
-
+  discard rlnIdentifier.copyFrom(proofBytes[nullifierOffset..rlnIdentifierOffset-1])
+  
   let output = RateLimitProof(proof: zkproof,
                               merkleRoot: proofRoot,
                               epoch: epoch,
                               shareX: shareX,
                               shareY: shareY,
-                              nullifier: nullifier)
+                              nullifier: nullifier,
+                              rlnIdentifier: rlnIdentifier)
 
   return ok(output)
 
 proc serialize(proof: RateLimitProof, data: openArray[byte]): seq[byte] =
   ## a private proc to convert RateLimitProof and data to a byte seq
   ## this conversion is used in the proof verification proc
-  ## the order of serialization is based on https://github.com/kilic/rln/blob/7ac74183f8b69b399e3bc96c1ae8ab61c026dc43/src/public.rs#L205
-  ## [ proof<256>| root<32>| epoch<32>| share_x<32>| share_y<32>| nullifier<32> | signal_len<8> | signal<var> ]
+  ## [ proof<128> | root<32> | epoch<32> | share_x<32> | share_y<32> | nullifier<32> | rln_identifier<32> | signal_len<8> | signal<var> ]
   let lenPrefMsg = appendLength(@data)
   var proofBytes = concat(@(proof.proof),
                           @(proof.merkleRoot),
                           @(proof.epoch),
                           @(proof.shareX),
```

Comment:
This order of share x / y / nullifier makes more sense to me

Reply:
See https://github.com/status-im/nwaku/pull/1030/files/765e8c5fa6457beb0b878da18a98682c935ac9ac#r914918440